documentation/source/coroutines.rst

```diff
@@ -117,71 +87,36 @@ Coroutines can be joined to end parallel operation within a function.
             else:
                 break
 
-Coroutines can be killed before they complete, forcing their completion before
+Forked coroutines can be killed before they complete, forcing their completion before
 they'd naturally end.
 
 .. code-block:: python3
 
     @cocotb.test()
-    def test_different_clocks(dut):
+    async def test_different_clocks(dut):
         clk_1mhz   = Clock(dut.clk, 1.0, units='us')
         clk_250mhz = Clock(dut.clk, 4.0, units='ns')
 
         clk_gen = cocotb.fork(clk_1mhz.start())
         start_time_ns = get_sim_time(units='ns')
-        yield Timer(1, units='ns')
-        yield RisingEdge(dut.clk)
+        await Timer(1, units='ns')
+        await RisingEdge(dut.clk)
         edge_time_ns = get_sim_time(units='ns')
-        # NOTE: isclose is a Python 3.5+ feature
         if not isclose(edge_time_ns, start_time_ns + 1000.0):
             raise TestFailure("Expected a period of 1 us")
 
-        clk_gen.kill()
+        clk_gen.kill()  # kill clock coroutine here
 
         clk_gen = cocotb.fork(clk_250mhz.start())
         start_time_ns = get_sim_time(units='ns')
-        yield Timer(1, units='ns')
-        yield RisingEdge(dut.clk)
+        await Timer(1, units='ns')
+        await RisingEdge(dut.clk)
         edge_time_ns = get_sim_time(units='ns')
-        # NOTE: isclose is a Python 3.5+ feature
         if not isclose(edge_time_ns, start_time_ns + 4.0):
             raise TestFailure("Expected a period of 4 ns")
 
-.. _async_functions:
-
-Async functions
-===============
-
-Python 3.5 introduces :keyword:`async` functions, which provide an alternative
-syntax. For example:
-
-.. code-block:: python3
-
-    @cocotb.coroutine
-    async def wait_10ns():
-        cocotb.log.info("About to wait for 10 ns")
-        await Timer(10, units='ns')
-        cocotb.log.info("Simulation time has advanced by 10 ns")
-
-To wait on a trigger or a nested coroutine, these use :keyword:`await` instead
-of :keyword:`yield`. Provided they are decorated with ``@cocotb.coroutine``,
-``async def`` functions using :keyword:`await` and regular functions using
-:keyword:`yield` can be used interchangeable - the appropriate keyword to use
-is determined by which type of function it appears in, not by the
-sub-coroutine being called.
-
-.. versionadded:: 1.4
-    The :any:`cocotb.coroutine` decorator is no longer necessary for ``async def`` coroutines.
-    ``async def`` coroutines can be used, without the ``@cocotb.coroutine`` decorator, wherever decorated coroutines are accepted,
-    including :keyword:`yield` statements and :any:`cocotb.fork`.
-
-.. note::
-    It is not legal to ``await`` a list of triggers as can be done in
-    ``yield``-based coroutine with ``yield [trig1, trig2]``. Use
-    ``await First(trig1, trig2)`` instead.
```

Comment:
I think we should keep this note around somewhere too.

Reply:
It's in the documentation for `First`.